core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1129,9 +1129,7 @@ public void workerRegister(long workerId, List<String> storageTiers,
       throw new NotFoundException(ExceptionMessage.NO_WORKER_FOUND.getMessage(workerId));
     }
 
-    if (options.hasBuildVersion()) {
-      worker.setBuildVersion(options.getBuildVersion());
-    }
+    worker.setBuildVersion(options.getBuildVersion());
```

Comment:
Actually, there's a streaming register defined in `workerRegisterStream`. Does that have the same logic receiving version info?

Reply:
If streaming registration is missing this logic as a whole, you can add that in a separate PR.